one-step-theta.h

```diff
@@ -0,0 +1,110 @@
+#ifndef ONE_STEP_THETA_H
+#define ONE_STEP_THETA_H
+
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_gmres.h>
+
+using namespace dealii;
+
+template <typename Operator>
+class OneStepTheta
+{
+public:
+    using Number = double;
+    OneStepTheta(Operator &pde_operator, LinearAlgebra::distributed::Vector<Number> &solution_n);
+
+    template <typename VectorType>
+    void UpdateBuffers(VectorType &solution) const;
+
+    template <typename VectorType>
+    void perform_time_step(double const time_step_size,
+                           VectorType &next_solutions) const;
+
+    void
+    vmult(LinearAlgebra::distributed::Vector<Number> &dst,
+          const LinearAlgebra::distributed::Vector<Number> &src) const;
+
+    template <typename VectorType>
+    void righthandside(VectorType &right_hand_side,
+                       VectorType &current_solution) const;
```

Comment:
Do these functions need to be public?

Reply:
The righthandside not. The vmult must be public, because it is used inside the linear solver.